### PR TITLE
Telescience Rebalance

### PR DESCRIPTION
**Pull request by monster860**
Now, you don't have to rely on ANY external tools at all to be able to use this! Not dream maker, not that fancy telescience calculator, NOTHING!

Anyways, I'm considering having this replace normal teleporters.

Pics on the new teleport animation will arrive soon (TM)

:cl: monster860
rscadd: Telescience teleportation now has fancy animations, based off of abductor teleports, but they work on objects besides humans thanks to icon manipulation
tweak: Telescience now takes an entire 8 seconds to complete. The target's view will be at the destination, but will be inside an invulnerable effect object during that time. In other words, the target will have more than enough time to yell for "HALP!" over the radio. Also, the target will be able to resist out of the teleport, causing the teleporter beam to collapse, but dealing toxins damage.
tweak: Telescience destinations are now selected like this: Select a beacon object (Tracking beacon, tracking implant, and crew member with maxed suit sensors) from a list of objects, and you can now lock on to any object within a range determined by the amount of bluespace crystals. That object will be the coordinates used when hitting send or receive
rscadd: Telescience has an access log that can be viewed and used for evidence to prove that person A teleported person B into the singularity.
/:cl:

### CODE REVIEW

**Comment by LittleBigKid2000** [comment]
muh telescience cheese
:+1:

**Comment by theColdflame** [comment]
:-1: to being able to pull crewmembers from suit sensors, the victim has absolutely zero chance to defend themselves beyond uselessly calling for help

aside from that, these seem to be good changes

**Comment by LittleBigKid2000** [comment]
I think the target has to stay still for 8 whole seconds

**Comment by theColdflame** [comment]
'Telescience now takes an entire 8 seconds to complete. The target's view will be at the destination, but will be inside an invulnerable effect object during that time. In other words, the target will have more than enough time to yell for "HALP!" over the radio'

seems to imply you're teleported immediately but remain intangible for 8s, unless I'm reading it wrong

**Comment by TheDZD** [comment]
```
code/modules/telesci/telesci_computer.dm:92:warning: check_target_sanity: unused label
```

**Comment by monster860** [comment]
@theColdflame

Yes, but the victim can *PREPARE FOR BATTLE*. Get out that fire extinguisher he was hiding. And of course yell for help.

**Comment by Earthdivine** [comment]
I don't quite understand this all.

Is the selection changed to be "Any number of tiles in a direction around a Telebeacon based on the consoles telecrystal amount"; or is that just choosing where they are going to land? If it's the former I like that alot better then telescience of old since you have to set the beacons in hidden spots to grab the stuff you want rather then "Look at the code."

**Comment by monster860** [comment]
@Earthdivine Your selection is any movable object within a range of a beacon, where with 4 crystals, you can go 4 tiles away. However, a wall in the way = no tele for you.

**Comment by monster860** [comment]
Added resist, added cooldown, and fixed logs. Also fixed compile warning

File: code/modules/telesci/telesci_computer.dm [hunk 1]
```diff
@@ -301,6 +309,21 @@
...
 	qdel(src)
 
+/obj/effect/teleport_blink/container_resist(mob/living/M)
+	if(!reset_turf)
+		return
+	if(prob(50))
+		to_chat(M, "<span class='warning'>You resist the teleportation!</span>")
+		visible_message("<span class='warning'>The teleport beam collapses!</span>")
+		M.forceMove(reset_turf)
+		M.adjustToxLoss(15) // A bit of toxins damage.
```
Comment (FlattestGuitar) [inline code/modules/telesci/telesci_computer.dm]: I don't see why the toxin damage is necessary. That way you can just spam teleport a HoS and you'll either get them teleported into your non-descript maint cave or you'll just crit them with toxin damage if you try long enough.

Comment (monster860) [inline code/modules/telesci/telesci_computer.dm]: You're forgetting that the HoS is going to turn off his sensors

Comment (FlattestGuitar) [inline code/modules/telesci/telesci_computer.dm]: But why does it deal 15 toxin damage? It just makes no sense...

Also, you can do that to 20 different crewmembers who don't turn their suit sensors off. A pretty bad idea considering they most likely won't be able to tell where they were being teleported to, so you'll leave no trace.

**Comment by Aurorablade** [comment]
You need to remove the suit sensor bit too much room for abuse

**Comment by Spacemanspark** [comment]
My concern mirrors that of Coldflame and Aurorablade.
Remove that and I'm willing to try this.

**Comment by Lady-Luck** [comment]
It's not perfect, but it's far better than what we have now. Let's give it a shot.

File: code/modules/telesci/telesci_computer.dm (hunk 1 above)
Comment (Fox-McCloud) [inline code/modules/telesci/telesci_computer.dm]: I don't like the idea of it dealing damage either, to be frank.

**Comment by Fox-McCloud** [comment]
When you say suit sensors, do you mean anyone who has maxed suit sensors or only on that z-level?

**Comment by TheDZD** [comment]
Definitely concerned with the suit sensor thing, although it'd be pretty much useless if you could only lock onto things within a certain range of tracking implants beacons.

A few things you might want to specify in the PR description though,
- How exactly do you select objects within range of a tracking beacon/tracking implant? Is it some sort of dropdown list (such as when you use a verb that can target any atom within a certain range)?
- How much range does each telecrystal give?

That said, I think in its current state with the damage + RNG + suit sensors lock, it can be a bit BS to play against. Some ideas below:
- Make locks via suit sensors take longer/be cancelled by resisting 100% of the time.
- Have a 5-10 second cooldown between teleport attempts.

**Comment by monster860** [comment]
@TheDZD This already has a 10 second cooldown

**Comment by monster860** [comment]
@Fox-McCloud Anyone with maxed suit sensors on the z-level

**Comment by uraniummeltdown** [comment]
Can we not nerf everything that can be used by antags to make their lives a bit easier? I'm sick of these shitty nerfs which get merged because people keep crying that the antag actually accomplished his objective and killed you. First Genetics got nerfed (aka completely trashed) so hard that it's worse than shit, Genetics is completely useless now due to the absurd amount of time it takes and the RNG. Telescience is not roundstart, it requires diamonds and bluespace crystals. needs to be calibrated. Telenapping people requires them to have sensors enabled, standing still and on the same z-level as you. Telenapping a head is dangerous as the Death Alarm will reveal your lair, not to mention your victims spamming ;HELP MED MAINT into the radio. People are just mad they were killed by an antag using telescience rather than the antag risking their entire round trying to kill them and getting perma/execution while the target is cloned. Enough with the hugbox PRs, telescience is fine as it is. Do not merge. What's even stupider is the fact that walls block teleports with this new system, seriously?

**Comment by Doukhan** [comment]
Hugbox is life.
Triangles are where it's at.

**Comment by LittleBigKid2000** [comment]
Traitors, Changelings, Vampires, and Wizards should have no uplinks/abilities/spells and start in perma with prison clothes and a tracking implant because i ded. Also remove xenos and blobs and abductors and revenants. Remove all non-human species too.

**Comment by uraniummeltdown** [comment]
@monster860 unless that actually happens there's no need for this new system, I get that you worked hard on the code but we don't need another huge nerf out of the blue, I'm still mad about the removal of Genetics because Sec couldn't deal with the taser not being an "I win" button any more.

**Comment by Bram961** [comment]
God no, telescience doesnt need to be any easier to use which this does from what I can tell and the suit sensors thing will destroy antags. A single telescientist can nab away a target you're dealing damage to in seconds as long as they have their suit sensors on.

No, to this entire PR. I feel like there are much better things telescience could be changed into but this is not one of them, this actually makes it worse.

**Comment by monster860** [comment]
There's a really simple solution:

Turn your suit sensors *off*

**Comment by Spacemanspark** [comment]
Why though.

**Comment by Fox-McCloud** [comment]
No response or updates despite maintainers continued prodding--PR is stale, closing this.

**Comment by Iamgoofball** [comment]
Ruins the point of telescope entirely

**Comment by Lady-Luck** [comment]
>Pull Request opened by Fox McCloud: Fox McCloud - Removes telescience
3 seconds later:
Pull Request merged by TheDZD: Fox McCloud - Removes telescience

Go go go

**Pull request closed without merging**